### PR TITLE
Sign Artifacts in Github Action

### PR DESCRIPTION
**Pull request by affonsov**
### Summary

Sign Artifacts in Github Action

### Description

Added new parameter in the Gradle GitHub action that will allow signing the artifacts to be used in the maven repo.

### Additional Reviewers
@affonsoBQ
@alinaliBQ
@andiem-bq
@alexey-temnikov
@birschick-bq

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -27,6 +27,10 @@ on:
         description: Test without DocumentDB? (1/0)
         required: false
         default: '1'
+      mavenFilesPreparation:
+        description: Prepare files to publish in maven repo? (1/0)
+        required: false
+        default: '0'
```
Comment (alinaliBQ) [inline .github/workflows/gradle.yml:33]: default value is 0, which means we will not prepare files to publish by default?

Comment (affonsov) [reply .github/workflows/gradle.yml:33]: yes this does not need to be triggered by all prs. we  will trigger this on-demand when we have to create a release

**Review by alinaliBQ — APPROVED** [review]

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -69,6 +76,20 @@ jobs:
...
+        base64 -d secring.kbx.b64 > secring.kbx
+    - name: Build with Gradle with local and optional remote DocumentDB tests
+      if: ${{ github.event.inputs.mavenFilesPreparation }}
+      run: |
+        ./gradlew build publishToMavenLocal -Psigning.secretKeyRingFile=secring.kbx -Psigning.password=$GPG_KEY_PASSWORD -Psigning.keyId=$GPG_SIGN_KEY
+    - name: Copy Files for Maven
+      if: ${{ github.event.inputs.mavenFilesPreparation }}
+      run: | 
+        mkdir output/maven
+        cp -R ~/.m2/repository/software/amazon/documentdb/jdbc/documentdb-jdbc/* output/maven
```
Comment (birschick-bq) [inline .github/workflows/gradle.yml]: @affonsoBQ Did you resolve the issue of which outputs/JAR files to publish? Could it be fine-tuned here?

Comment (affonsov) [reply .github/workflows/gradle.yml]: fine-tuned to make the life easier in the manual publishing part

File: .github/workflows/gradle.yml [hunk 3]
```diff
@@ -69,13 +76,46 @@ jobs:
...
         mkdir output/reports
         cp build/libs/*.jar output/jarfile
         cp -R build/reports output/reports
+    - name: Copy Files for Maven
+      if: ${{ github.event.inputs.mavenFilesPreparation }}
+      run: | 
+        mkdir -p output/maven
+        name=$(ls build/libs | grep documentdb-jdbc.*-all.jar.asc)
+        vers=$(awk -F- '{print $3}' <<<"$name")
+        cp ~/.m2/repository/software/amazon/documentdb/jdbc/documentdb-jdbc/$vers/documentdb-jdbc-$vers.jar output/maven
```
Comment (birschick-bq) [inline .github/workflows/gradle.yml]: Readability could be improved if you assign a variable for the source path:
```
src_folder=~/.m2/repository/software/amazon/documentdb/jdbc/documentdb-jdbc/$vers
cp $src_folder/documentdb-jdbc-$vers.jar output/maven
...
```

File: .github/workflows/gradle.yml [hunk 4]
```diff
@@ -69,13 +76,47 @@ jobs:
...
+        ./gradlew build publishToMavenLocal -Psigning.secretKeyRingFile=secring.kbx -Psigning.password=$GPG_KEY_PASSWORD -Psigning.keyId=$GPG_SIGN_KEY
     - name: Copy Build
       run: |
         mkdir output
         mkdir output/jarfile
         mkdir output/reports
         cp build/libs/*.jar output/jarfile
         cp -R build/reports output/reports
+    - name: Copy Files for Maven
+      # if: ${{ github.event.inputs.mavenFilesPreparation }}
```
Comment (birschick-bq) [inline .github/workflows/gradle.yml]: Just checking - we do want to only run this step when preparing maven files, correct?

Comment (affonsov) [reply .github/workflows/gradle.yml]: I just commented to test in the PR run. I will uncomment soon

**Review by birschick-bq — APPROVED** [review]

**Pull request merged**